nilearn/maskers/nifti_labels_masker.py

```diff
@@ -664,12 +618,70 @@ def fit(
         # Infer the number of elements in the mask
         # This is equal to the number of unique values in the label image,
         # minus the background value.
-        self.n_elements_ = (
-            np.unique(get_data(self._resampled_labels_img_)).size - 1
-        )
+        self.n_elements_ = len(self.lut_) - 1
 
         return self
 
+    def _check_labels(self):
+        """Check labels.
+
+        - checks that labels is a list of strings.
+        """
+        labels = self.labels
+        if not isinstance(labels, list):
+            raise TypeError(
+                f"'labels' must be a list. Got: {type(labels)}",
+            )
+        if not all(isinstance(x, str) for x in labels):
+            types_labels = {type(x) for x in labels}
+            raise TypeError(
+                "All elements of 'labels' must be a string.\n"
+                f"Got a list of {types_labels}",
+            )
+
+    def _generate_lut(self):
```

Comment:
this and a few of the properties introduced in this PR should be refactored across volume and surface maskers: to be done in a follow up PR